### PR TITLE
Created Role Based Access System with MYSQL

### PR DESCRIPTION
**Pull request by vc-jinal**
Created table to store user informatio,roles table and permission table and to role_permission table 
Created api to get all permissions for a role.
Created api to add and remove permissions from role.(Create 2 separate api).
Created api to delete permission .
Created api to update permission name and role name.
Created api to delete role.
Created api to get role for user.
Created api to get permissions for a user.

### CODE REVIEW

File: role-based-access-system/src/app.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import express from "express";
+import route from "./routes/index.route.js";
```
Comment (gaurangviitorcloud) [inline role-based-access-system/src/app.js:2]: Are you sure this code is working cause i can't find index.js in routes folder anywhere

File: role-based-access-system/src/routes/permission.route.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+import { Router } from "express";
+const route = Router();
+import query from "../db/query.js";
+
+// get all permissions for a role.
+route.get("/roles/permissions/:roleId", async (req, res) => {
```
Comment (gaurangviitorcloud) [inline role-based-access-system/src/routes/permission.route.js:6]: Why role route is in permissions routes file ?

File: role-based-access-system/src/routes/permission.route.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        }
+        return res.send({ statusCode: 200, message: "Permission Deleted", data: deletePermission })
+    }
+    catch (error) {
+        return res.send({ statusCode: 500, message: "Internal Server Error" })
+    }
+})
+
+//update permission name 
+route.put('/update/permission/:permissionId', async (req, res) => {
```
Comment (gaurangviitorcloud) [inline role-based-access-system/src/routes/permission.route.js:67]: Route should not start with update

File: role-based-access-system/src/routes/role.route.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import { Router } from "express";
+const route = Router();
+import query from "../db/query.js";
+import connection from "./db/connection.js";
+
+//update role name
+route.put('roles/:roleId', async (req, res) => {
```
Comment (gaurangviitorcloud) [inline role-based-access-system/src/routes/role.route.js:7]: Are you sure this route is working ?

File: role-based-access-system/src/routes/user.route.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import { Router } from "express";
+const route = Router();
+import query from "../db/query.js";
+
+//get role for user
+route.get('/role/user/:userId', async (req, res) => {
```
Comment (gaurangviitorcloud) [inline role-based-access-system/src/routes/user.route.js:6]: Why role route in user.routes.js ?

File: role-based-access-system/src/routes/user.route.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        }
+        return res.send({ statusCode: 200, message: "Get Role of User", data: getUserRole })
+    }
+    catch (error) {
+        return res.send({ statusCode: 500, message: "Internal Server Error" });
+    }
+})
+
+//get permission for user
+route.get('/permission/user/:userId', async (req, res) => {
```
Comment (gaurangviitorcloud) [inline role-based-access-system/src/routes/user.route.js:25]: Why permission route in user.routes.js ?

File: role-based-access-system/src/routes/permission.route.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        }
+        return res.send({ statusCode: 200, message: "Added permission to Role", data: permission })
+    }
+    catch (error) {
+        return res.send({ statusCode: 500, message: "Internal Server Error" });
+    }
+})
+
+//delete permission of role
+route.delete('/delete/permission/:roleId/:permissionId', async (req, res) => {
```
Comment (gaurangviitorcloud) [inline role-based-access-system/src/routes/permission.route.js:36]: Route should not start with delete

File: role-based-access-system/src/routes/permission.route.js [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+            return res.send({ statusCode: 404, message: "Data not found" });
+        }
+        return res.send({ statusCode: 200, message: "Permissions fetched successfully", data: permission });
+    } catch (error) {
+        return res.send({ statusCode: 500, message: "Internal Server Error" });
+    }
+});
+
+//add permission to role
+route.post('/add/permission/:roleId/:permissionId', async (req, res) => {
```
Comment (gaurangviitorcloud) [inline role-based-access-system/src/routes/permission.route.js:20]: Route should not start with add

**Review by gaurangviitorcloud — CHANGES_REQUESTED** [review]
- Please check if your code is working before raising a PR . Don't do task for sake of doing it.
- You should not use "add" , "delete" ,"update" in route for that we have HTTP verbs please google it.

File: role-based-access-system/src/.gitignore
Comment (vc-mohit) [inline role-based-access-system/src/.gitignore:1]: This is required.

File: role-based-access-system/src/routes/permission.route.js (hunk 2 above)
Comment (gaurangviitorcloud) [reply role-based-access-system/src/routes/permission.route.js:6]: This is not fixed yet roles api should not be in permissions routes

File: role-based-access-system/src/routes/permission.route.js [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+        }
+        return res.send({ statusCode: 200, message: "Permission Deleted", data: deletePermission })
+    }
+    catch (error) {
+        return res.send({ statusCode: 500, message: "Internal Server Error" })
+    }
+})
+
+//update permission name 
+route.put('/permission/:permissionId/:permissionName', async (req, res) => {
```
Comment (gaurangviitorcloud) [inline role-based-access-system/src/routes/permission.route.js:88]: Permission name should be in request body

File: role-based-access-system/src/routes/roles.route.js [hunk 10]
```diff
@@ -0,0 +1,56 @@
+import { Router } from "express";
+const route = Router();
+import query from "../db/query.js";
+
+//update role name
+route.put('/roles/:roleId/:roleName', async (req, res) => {
```
Comment (gaurangviitorcloud) [inline role-based-access-system/src/routes/roles.route.js:6]: Role name should be in request.body

**Review by gaurangviitorcloud — CHANGES_REQUESTED** [review]
- Please fix routes according to proper naming convention.

**Comment by gaurangviitorcloud** [comment]
re raise with proper changes

File: role-based-access-system/src/routes/permission.route.js (hunk 2 above)
Comment (vc-jinal) [reply role-based-access-system/src/routes/permission.route.js:6]: Changed role api to role route

File: role-based-access-system/src/routes/permission.route.js (hunk 3 above)
Comment (vc-jinal) [reply role-based-access-system/src/routes/permission.route.js]: removed it

File: role-based-access-system/src/routes/user.route.js (hunk 6 above)
Comment (vc-jinal) [reply role-based-access-system/src/routes/user.route.js]: made chnages in api url

File: role-based-access-system/src/routes/permission.route.js (hunk 7 above)
Comment (vc-jinal) [reply role-based-access-system/src/routes/permission.route.js]: removed delete

File: role-based-access-system/src/.gitignore
Comment (vc-jinal) [reply role-based-access-system/src/.gitignore:1]: removed it

File: role-based-access-system/src/app.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import express from "express";
+import permission from "./routes/permission.route.js";
```
[the remark this answers is not in the document]
Comment (vc-jinal) [reply role-based-access-system/src/app.js:2]: fix it

**Pull request closed without merging**